src/watchdog/src/watchdog.py

```diff
@@ -241,11 +241,17 @@ def _map_fn(item):
     map(_map_fn, podsJsonObject["items"])
 
 
-def collect_healthz(gauge, histogram, service_name, address, port, url):
+def collect_healthz(gauge, histogram, service_name, address, port, url, tls, ca_path, bearer_path):
     with histogram.time():
         error = "ok"
         try:
-            error = requests.get("http://{}:{}{}".format(address, port, url)).text
+            if tls:
+                with open(bearer_path, 'r') as bearer_file:
```

Comment:
Please, read the bearer in main, and pass content to `collect_healthz` function.

If you implement this way, it will read the file many times in runtime, which incur IO.

Reply:
pass content